ecr/resolver.go

```diff
@@ -203,29 +214,62 @@ func (r *ecrResolver) getClient(region string) ecrAPI {
 	return r.clients[region]
 }
 
-type manifestContent struct {
-	SchemaVersion int64         `json:"schemaVersion"`
-	Signatures    []interface{} `json:"signatures,omitempty"`
-	MediaType     string        `json:"mediaType,omitempty"`
+// manifestProbe provides a structure to parse and then probe a given manifest
+// to determine its mediaType.
+type manifestProbe struct {
+	// SchemaVersion is version identifier for the manifest schema used.
+	SchemaVersion int64 `json:"schemaVersion"`
+	// Explicit MediaType assignment for the manifest.
+	MediaType string `json:"mediaType,omitempty"`
+	// Docker Schema 1 signatures.
+	Signatures []json.RawMessage `json:"signatures,omitempty"`
+	// OCI or Docker Manifest Lists, the list of descriptors has mediaTypes
+	// embedded.
+	Manifests []json.RawMessage `json:"manifests,omitempty"`
 }
 
+// TODO: make this error and handle in caller.
 func parseImageManifestMediaType(ctx context.Context, body string) string {
-	var manifest manifestContent
+	// The unsigned variant of Docker v2 Schema 1 manifest mediaType.
+	const mediaTypeDockerSchema1ManifestUnsigned = "application/vnd.docker.distribution.manifest.v1+json"
+
+	// The type used as a fallback when parsing is not feasible.
+	//
+	// TODO: confirm this as the correct fallback mediaType.
```

Comment:
I don't have concrete data, but I believe this matches the majority of images in the wild, as that's what Docker generates.

Reply:
Other than the majority case - is it correct to bail and return the Docker media type when the parser can't unmarshal JSON? 

It seems sort of odd, to me anyway, that the valid-JSON fallthrough case results in `""` while JSON parse errors results in a statistically likely media type.

Reply:
> It seems sort of odd, to me anyway, that the valid-JSON fallthrough case results in "" while JSON parse errors results in a statistically likely media type.

Ah, I see what you're saying.  I'd bias toward keeping the behavior that we have right now, and then following up with a separate PR to adjust the behavior and make it less odd.  (Realistically, it might be better to return a `(string, error)` and make both of these cases explicit errors.)

Reply:
Created an issue (#18) to track follow up work.